### ISSUE TITLE
Incorrect Handling of Multiple DOMElement Objects in source_block_raw Method

### ISSUE THREAD

**Issue by veerap000** [Automattic/vip-block-data-api#81]
I found an issue with the `source_block_raw` method in [/src/parser/content-parser.php](https://github.com/Automattic/vip-block-data-api/blob/trunk/src/parser/content-parser.php#L677).

### Current Behavior:

In the current implementation, the function only returns the first matched DOMElement when there are multiple elements within the _core/html_ block. Specifically, it uses:
`$attribute_value = trim( $crawler->outerHtml() );`

This causes a problem when the _core/html_ block contains multiple DOMElement objects. For example, consider the following HTML content:
`<h2>My HTML heading</h2><p>My HTML paragraph</p>`

Currently, only the first element (`<h2>My HTML heading</h2>`) is returned, while the rest of the elements are ignored.

### Proposed Solution:

To address this, I propose modifying the function to iterate over all matched elements, collect their outer HTML, and then concatenate them together. This can be accomplished with the following change:

```
$attribute_values = $crawler->each(function ($node) {
    return trim( $node->outerHtml() );
});

$attribute_value = trim( implode( '', $attribute_values ) );
```

This approach ensures that all DOMElement objects within the core/html block are captured and returned correctly.

**Example Output:**
With this change, the example input:
`<h2>My HTML heading</h2><p>My HTML paragraph</p>`
Will be correctly handled, returning both elements as expected, instead of only the first.

Please let me know if this proposed fix is suitable or if further adjustments are needed.

**Comment by alecgeatches** [comment]
@veerap000 Thank you for the report! I think your problem explanation and solution look right to me. We'll see if we can get this fixed shortly.

**Comment by alecgeatches** [comment]
@veerap000 Merged a fix over in https://github.com/Automattic/vip-block-data-api/pull/82! We used a slightly different approach:

https://github.com/Automattic/vip-block-data-api/blob/a4939bbe953548544e670e74207d1973d1ab4ae0/src/parser/content-parser.php#L685-L694

I noticed during testing that the `trim()` + `implode()` method would remove inner whitespace, which can be important for text like this:

```html
<span>These</span> <span class="highlight">spaces</span> <span>are</span> <span>important</span>.
```

The new method still trims the outer string but will retain inner whitespace, including things like [newlines and non-breaking spaces](https://github.com/Automattic/vip-block-data-api/blob/a4939bbe953548544e670e74207d1973d1ab4ae0/tests/parser/sources/test-source-raw.php#L121-L122).

Thank you for reporting this! We have a small test fix to do, and should have a new release version out shortly.
